File: cam-srv/camera-image-loader.js
const axios = require('axios');

class CameraImageLoader {
    static loadByCamera(camera) {
        return this.loadById(camera.id);
    }

    static loadById(id) {
        return axios.get(this.getUrl(id), {
            responseType: 'arraybuffer',
            headers: {
                'Referer': this.getReferer()
            },
        })
            .then(this.parseResponse)
            .then(img => {
                img.id = id;
                return img;
            });
    }

    static parseResponse(response) {
        const data = response.data;
        const date = response.headers['last-modified'];

        return {
            data: data,
            date: new Date(date)
        }
    }

    static getUrl(id) {
        return `http://www.svz-bw.de/kamera/ftpdata/${id}/${id}_gross.jpg`;
    }

    static getReferer() {
        return 'https://www.svz-bw.de';
    }
}

module.exports = CameraImageLoader;

File: android/app/src/main/java/com/momo5502/stauanalyse/camera/CameraImageLoader.java
package com.momo5502.stauanalyse.camera;

import com.momo5502.stauanalyse.util.Callback;
import com.momo5502.stauanalyse.util.Downloader;

import java.util.HashMap;

public class CameraImageLoader {
    private Downloader downloader = new Downloader();

    public void get(String id, Callback<byte[]> callback) {
        HashMap<String, String> headers = new HashMap<>();
        headers.put("Referer", "https://www.svz-bw.de");

        downloader.download(callback, getUrl(id), headers);
    }

    private String getUrl(String camera) {
        return "https://www.svz-bw.de/kamera/ftpdata/" + camera + "/" + camera + "_gross.jpg";
    }
}
